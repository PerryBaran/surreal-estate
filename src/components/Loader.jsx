import React from "react";
import PropTypes from "prop-types";
import { ClipLoader } from "react-spinners";
import css from "../styles/loader.module.css";

const Loader = ({ loading, size }) => {
  return (
    <div className={css.loader}>
      <ClipLoader
        color="#a20aa2"
        loading={loading}
        size={size}
        aria-label="Loading Spinner"
        data-testid="loader"
      />
    </div>
  );
};

Loader.defaultProps = {
  size: 25,
};

Loader.propTypes = {
  loading: PropTypes.bool.isRequired,
  size: PropTypes.number,
};

export default Loader;
