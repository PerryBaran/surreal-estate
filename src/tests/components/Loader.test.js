import React from "react";
import { render } from "@testing-library/react";
import Loader from "../../components/Loader";

describe("Loader", () => {
  const validProps = {
    loading: true,
    size: 40,
  };

  test("snapshot", () => {
    const { asFragment } = render(
      <Loader loading={validProps.loading} size={validProps.size} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
