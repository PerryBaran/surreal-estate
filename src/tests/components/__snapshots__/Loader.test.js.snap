// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    <span
      aria-label="Loading Spinner"
      data-testid="loader"
      style="width: 40px; height: 40px; border-radius: 100%; border: 2px solid; border-top-color: #a20aa2; border-bottom-color: transparent; border-left-color: #a20aa2; border-right-color: #a20aa2; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
    />
  </div>
</DocumentFragment>
`;
